### PR TITLE
Update workflow versions, platforms, library and tool versions + README

### PR DESCRIPTION
**Pull request by felixdivo**
I think most changes do not require explanation. The README was changed since Python 3.6 reached EOL as discussed in (the end of the discussion of) #528.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -73,7 +72,7 @@
     version=version,
     packages=find_packages(exclude=["test*", "doc", "scripts", "examples"]),
     scripts=list(filter(isfile, (join("scripts/", f) for f in listdir("scripts/")))),
-    author="Python CAN contributors",
+    author="python-can contributors",
```
Comment (felixdivo) [inline setup.py:75]: I renamed this since it's more familiar for most, I suppose?

**Review by hardbyte — APPROVED** [review]

**Comment by felixdivo** [comment]
@zariiii9003 Do you approve?

**Review by zariiii9003 — APPROVED** [review]
> @zariiii9003 Do you approve?

Yeah i was just a little hesitant to drop 3.6 already.

**Comment by felixdivo** [comment]
> Yeah i was just a little hesitant to drop 3.6 already.

I think there is no point in maintaining an EOL version. People can still install it at their own risk. In the worst case, they have to stick to a certain commit. 🤷

**Pull request merged**